src/lib/error.pl

```diff
@@ -82,6 +82,9 @@
         ;   true
         ).
 must_be_(list, Term)    :- check_(error:ilist, list, Term).
+must_be_(list(Elem), Term) :- 
```

Comment:
This is very hard to implement correctly and may lead errors in the wrong order. For instance, a type&nbsp;error gives more information than an instantiation error and should be thrown if both errors can be thrown. Is that the case in all situations?

For these reasons, `list(...)` is not implemented in this library.

Reply:
I had already undone this change in the latest commit at the time of this review.